php/WP_CLI/Runner.php

```diff
@@ -326,8 +325,18 @@ public function run_command( $args, $assoc_args = array(), $options = array() )
 		}
 	}
 
-	private function _run_command() {
+	private function _run_command_and_exit() {
 		$this->run_command( $this->arguments, $this->assoc_args );
+		if ( $this->cmd_starts_with( array( 'help' ) ) ) {
+			// Help couldn't find the command so exit with suggestion.
+			$possible_suggestion = $this->find_command_to_run( array_slice( $this->arguments, 1 ) );
+			if ( is_string( $possible_suggestion ) ) {
+				WP_CLI::error( $possible_suggestion );
+			}
+			// Should never happen.
```

Comment:
Well if it does find the command then it doesn't return...if it doesn't then it does...then `find_command_to_run()` runs the same loop as `find_command_to_help()` so it will fail too and return a suggestion...100%!